src/tests/JIT/Performance/JittedMethodsCountingTest/JittedMethodsCountingTest.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+public class JittedMethodsCountingTest
+{
+    private const int MAX_JITTED_METHODS_ACCEPTED = 50;
+
+    [Fact]
+    public static int TestEntryPoint()
+    {
+        string testAppFolder = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        
+        string appName = Path.Combine(testAppFolder, "HelloWorld.dll");
+        string jitOutputFile = Path.Combine(testAppFolder, "jits.txt");
+
+        int appResult = RunHelloWorldApp(appName, jitOutputFile);
+        int jits = GetNumberOfJittedMethods(jitOutputFile);
+
+        return jits > 0 && jits <= MAX_JITTED_METHODS_ACCEPTED ? 100 : 101;
+    }
+
+    private static int RunHelloWorldApp(string appName, string jitOutputFile)
+    {
+        int appExitCode = -1;
+        string osExeSuffix = (OperatingSystem.IsWindows() ? ".exe" : "");
+        string coreRootPath = Environment.GetEnvironmentVariable("CORE_ROOT");
+        string coreRunPath = Path.Combine(coreRootPath, "corerun" + osExeSuffix);
+
+        using (Process app = new Process())
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = coreRunPath,
+                Arguments = appName,
+                CreateNoWindow = true,
+                UseShellExecute = false,
+            };
+
+            Console.WriteLine("Launching: {0} {1}", startInfo.FileName, startInfo.Arguments);
+
+            startInfo.EnvironmentVariables.Add("DOTNET_JitDisasmSummary", "1");
+            startInfo.EnvironmentVariables.Add("DOTNET_TieredCompilation", "0");
+            startInfo.EnvironmentVariables.Add("DOTNET_ReadyToRun", "1");
+            startInfo.EnvironmentVariables.Add("DOTNET_JitStdOutFile", jitOutputFile);
```

Comment:
I tried this without the stdout file redirection and it seemed to work for me when I specified <code>RedirectStandardOutput=true</code> in the <code>ProcessStartInfo</code> constructor and I used <code>app.StandardOutput.ReadToEnd()</code> before the call to <code>WaitForExit</code>. Having said that, using the <code>DOTNET</code> variable for explicit specification of JIT standard output seems easier at the very least because the app emits the <code>Hello world!</code> string at the last line and you'd need to modify the parser to cope with that.

Reply:
I think using `DOTNET_JitStdOutFile` is the better way for these main reasons:
- It's self-explanatory as to what we're doing with that file.
- We don't have to parse the app's output out of the way, plus only having the jitted methods can make investigations easier.
- The fact dotnet keeps eating the output on my machine is concerning. There might be some underlying bug that only seems to repro under specific circumstances, and it makes this test give incorrect results when it happens.